src/main/scala/viper/gobra/frontend/Parser.scala

```diff
@@ -671,9 +676,10 @@ object Parser {
           case spec ~ cond ~ body => PForStmt(None, cond, None, spec, body)
         }
 
-
     lazy val loopSpec: Parser[PLoopSpec] =
-      ("invariant" ~> expression <~ eos).* ^^ PLoopSpec
+      ("invariant" ~> expression <~ eos).* ~ ("decreases" ~> terminationMeasure <~ eos).? ^^ {
+        case invariants ~ terminationMeasure => PLoopSpec(invariants, terminationMeasure)
+      }
```

Comment:
I think this could be simplified to:
```suggestion
      ("invariant" ~> expression <~ eos).* ~ ("decreases" ~> terminationMeasure <~ eos).? ^^ PLoopSpec
```